modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/action/EditFolderMVCActionCommand.java

```diff
@@ -245,6 +259,75 @@ private void _updateFolder(ActionRequest actionRequest)
 		}
 	}
 
+	private void _updatePropagationPermission(
+			ActionRequest actionRequest, Folder folder,
+			ServiceContext serviceContext)
+		throws PortalException {
+
+		if (folder == null) {
+			return;
+		}
+
+		DLFolder dlParentFolder = _dlFolderLocalService.fetchDLFolder(
+			folder.getParentFolderId());
+
+		if ((dlParentFolder == null) ||
+			!dlParentFolder.isPermissionPropagationEnabled()) {
+
+			return;
+		}
+
+		Portlet portlet = _portletLocalService.getPortletById(
+			serviceContext.getCompanyId(), serviceContext.getPortletId());
+
+		PermissionPropagator permissionPropagator =
+			portlet.getPermissionPropagatorInstance();
+
+		if (permissionPropagator == null) {
+			return;
+		}
+
+		int[] roleTypes = RoleConstants.TYPES_REGULAR_AND_SITE;
+
+		Group group = _groupLocalService.fetchGroup(
+			dlParentFolder.getGroupId());
+
+		if (group != null) {
+			if (group.isDepot()) {
+				roleTypes = new int[] {
+					RoleConstants.TYPE_REGULAR, RoleConstants.TYPE_DEPOT
+				};
+			}
+			else {
+				Group parentGroup = _groupLocalService.fetchGroup(
+					group.getParentGroupId());
+
+				Predicate<Group> organizationGroupPredicate = g ->
+					(g != null) &&
+					(g.isOrganization() || g.isCompany() || g.isUser() ||
+					 g.isUserGroup());
+
+				if (organizationGroupPredicate.test(parentGroup) ||
+					organizationGroupPredicate.test(group)) {
+
+					roleTypes =
+						RoleConstants.TYPES_ORGANIZATION_AND_REGULAR_AND_SITE;
+				}
```

Comment:
I think the roleTypes should be RoleConstants.TYPES_REGULAR if the site is site of company, userGroup and User.